modules/consensus/synchronize_test.go

```diff
@@ -225,3 +225,69 @@ func TestBlockHistory(t *testing.T) {
 		}
 	}
 }
+
+// TestSendBlocksBroadcasts tests that the SendBlocks RPC call only Broadcasts
+// one block, no matter how many blocks are sent. In the case 0 blocks are
+// sent, tests that Broadcast is never called.
```

Comment:
you should also test the case where the consensus set is not extended, as no blocks should be broadcast in that event.